src/api/utils.js

```diff
@@ -225,12 +225,13 @@ export const getRestrictedOrgUnits = (orgUnits, orgUnitType) => {
     })
 }
 
-export const appendUsernameToDisplayName = userModelCollection => {
-    userModelCollection.forEach(user => {
-        user.displayName += ` (${user.userCredentials.username})`
+export const appendUsernameToDisplayName = userModelCollection =>
+    userModelCollection.toArray().map(userModel => {
+        const username = userModel.userCredentials.username
+        const user = userModel.toJSON()
+        user.displayName += ` (${username})`
```

Comment:
Interestingly `userModel.userCredentials.username` has a value, but `user.userCredentials.username` does not. I suspect this is a bug in the `toJSON` method on the `ModelBase` class in `d2`, but haven't spent much time looking into this, since we're moving away from `d2` anyway.

Reply:
I think it might be worth investigating here, who knows what else might be dropped when using `toJSON`

Reply:
I don't think this is required. This helper is only used by the GroupEditor on the UserGroupForm, and ultimately to update the user-group I just need to pass some ids. So whatever is dropped, I don't know, but I am 100% sure that I still have all the properties I need.